src/bundle.c

```diff
@@ -1095,16 +1093,7 @@ gboolean resign_bundle(RaucBundle *bundle, const gchar *outpath, GError **error)
 		manifest = loaded_manifest;
 	}
 
-	if (manifest->bundle_format == R_MANIFEST_FORMAT_PLAIN) {
-		g_print("Reading bundle in 'plain' format\n");
-	} else if (manifest->bundle_format == R_MANIFEST_FORMAT_VERITY || manifest->bundle_format == R_MANIFEST_FORMAT_CRYPT) {
-		g_print("Reading bundle in '%s' format\n", manifest->bundle_format == R_MANIFEST_FORMAT_VERITY ? "verity" : "crypt");
-		g_assert(bundle->size > (goffset)manifest->bundle_verity_size);
-	} else {
-		g_error("unsupported bundle format");
-		res = FALSE;
-		goto out;
-	}
+	g_print("Resigning '%s' format bundle\n", r_manifest_bundle_format_to_str(manifest->bundle_format));
```

Comment:
Why is the check for unsupported bundle formats not needed anymore? Perhaps `check_allowed_bundle_format` should also reject unknown formats, then we know after `check_bundle`, that the format is supported.

Reply:
It was a sanity check only, leading to a hard abort (notice it uses `g_error()`).

No valid code path should allow ending up with an unknown bundle format.
The reason we had it in there was mainly to have an `else` branch documented.

I can also put it into the string conversion if you like.

Reply:
Ah, the reason we can't error out here is that `parse_manifest()` already rejects any unsupported format.

Reply:
"Can't" as in "shouldn't if the parsing code remains valid, yes". This is how I understand the general usage of `g_error()`.